factory/processComponents.go

```diff
@@ -802,15 +798,15 @@ func (pcf *processComponentsFactory) indexGenesisBlocks(genesisBlocks map[uint32
 		return err
 	}
 
-	if pcf.outportHandler.HasDrivers() {
+	if pcf.statusComponents.OutportHandler().HasDrivers() {
```

Comment:
maybe add a TODO to rework these calls as they are pretty ugly and prone to errors. We should check if we have drivers in the SaveBlock function, hidden from the rest of the implementation.

Reply:
You right is not a very nice solution, but in order to can skip the preprocess part of the information that needs to be indexed, in case of no drivers we need this method.